### PR TITLE
add groups cli selectors

### PR DESCRIPTION
**Pull request by ryanfkeepers**
populates all selectors in the groups cli.  adds
both info-based filters (such as message creation
time) and also basic channel and mesage selection.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)

* #3989

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/cli/backup/groups.go [hunk 1]
```diff
@@ -36,26 +36,25 @@ const (
...
 const (
-	groupsServiceCommandCreateExamples = `# Backup all Groups data for Alice
-corso backup create groups --group alice@example.com 
+	groupsServiceCommandCreateExamples = `# Backup all Groups and Teams data for the Marketing group
+corso backup create groups --group Marketing
 
-# Backup only Groups contacts for Alice and Bob
-corso backup create groups --group engineering,sales --data contacts
+# Backup only Teams conversations messages
+corso backup create groups --group Marketing --data messages
```
Comment (neha-Gupta1) [inline src/cli/backup/groups.go:44]: Are `names` supported for groups? I am able to work only by providing `ID` with current main.

Comment (ryanfkeepers) [reply src/cli/backup/groups.go:44]: They are not supported at this time.  I'll be investigating this shortly.

Comment (ryanfkeepers) [reply src/cli/backup/groups.go:44]: @neha-Gupta1 
that support has been added.  See: https://github.com/alcionai/corso/pull/4242

**Review by vkamra — APPROVED** [review]

File: src/cli/backup/groups.go (hunk 1 above)
Comment (neha-Gupta1) [reply src/cli/backup/groups.go:44]: Perfect

**Pull request merged**